crates/astria-sequencer/src/app/tests_execute_transaction.rs

```diff
@@ -1042,3 +1042,38 @@ async fn app_execute_transaction_bridge_lock_unlock_action_ok() {
         "bridge should've transferred out whole balance"
     );
 }
+
+#[tokio::test]
+async fn transaction_execution_records_fee_event() {
+    let mut app = initialize_app(None, vec![]).await;
+
+    // transfer funds from Alice to Bob
+    let (alice_signing_key, _) = get_alice_signing_key_and_address();
+    let bob_address = address_from_hex_string(BOB_ADDRESS);
+    let value = 333_333;
+    let tx = UnsignedTransaction {
+        params: TransactionParams::builder()
+            .nonce(0)
+            .chain_id("test")
+            .build(),
+        actions: vec![
+            TransferAction {
+                to: bob_address,
+                amount: value,
+                asset: get_native_asset().clone(),
+                fee_asset: get_native_asset().clone(),
+            }
+            .into(),
+        ],
+    };
+
+    let signed_tx = Arc::new(tx.into_signed(&alice_signing_key));
+
+    let native_asset = get_native_asset();
+
+    let events = app.execute_transaction(signed_tx).await.unwrap();
+    let transfer_fee = app.state.get_transfer_base_fee().await.unwrap();
+    let fee_event =
+        crate::state_ext::construct_tx_fee_event(native_asset, transfer_fee, "Transfer");
```

Comment:
We could make `construct_tx_fee_event` private if it weren't for using it here.

It's probably about 50-50, but I think I'd have been inclined to favour keeping the function in the production code private and changing the test to something like:
```rust
    let event = events.first().unwrap();
    assert_eq!(event.kind, "tx.fees");
    assert_eq!(
        event.attributes[0],
        ("asset", get_native_asset().to_string()).index().into()
    );
    assert_eq!(
        event.attributes[1],
        ("feeAmount", transfer_fee.to_string()).index().into()
    );
    assert_eq!(
        event.attributes[2],
        ("actionType", "Transfer").index().into()
    );
```